examples/json.rs

```diff
@@ -46,86 +47,83 @@ fn parser() -> impl Parser<char, Json, Error = Simple<char>> {
                 .or(just('r').to('\r'))
                 .or(just('t').to('\t'))
                 .or(just('u').ignore_then(
-                    filter(|c: &char| c.is_digit(16))
+                    any()
+                        .filter(|c: &char| c.is_digit(16))
                         .repeated()
                         .exactly(4)
-                        .collect::<String>()
-                        .validate(|digits, span, emit| {
+                        .map_slice(|x| x)
+                        .try_map(|digits, span| {
+                            char::from_u32(u32::from_str_radix(digits, 16).unwrap())
+                                .ok_or_else(|| Rich::expected_found(None, None, span))
+                        })
+                        .recover_with(empty().to('\u{FFFD}')) // unicode replacement character
```

Comment:
I don't think this works? If parsing `\uFFFF` or some other invalid character, this will reset the parser position to the `u` rather than the end of the thing.

Reply:
Hmm, you're right. I think we may be far enough along to actually implement `validate` now, since there's at least an error trait and at least minimal error prioritization again, so I might actually do that since that's really what this wants anyways.